packages/trader/src/App/Components/Elements/ContractDrawer/contract-drawer.jsx

```diff
@@ -83,128 +60,107 @@ class ContractDrawer extends React.Component {
-        }
     };
 
-    render() {
-        if (!this.props.contract_info) return null;
-
-        // For non-binary contract, the status is always null, so we check for is_expired in contract_info
-        const fallback_result = this.props.contract_info.is_expired;
-
-        const { exit_tick_display_value, is_sold } = this.props.contract_info;
-
-        const { contract_info, contract_update_history, is_dark_theme, is_multiplier, toggleHistoryTab } = this.props;
-
-        const exit_spot = isUserSold(contract_info) && !is_multiplier ? '-' : exit_tick_display_value;
-
-        const contract_audit = (
-            <ContractAudit
-                contract_info={contract_info}
-                contract_update_history={contract_update_history}
-                contract_end_time={getEndTime(contract_info)}
-                is_dark_theme={is_dark_theme}
-                is_multiplier={is_multiplier}
-                is_open={true}
-                is_shade_visible={this.handleShade}
-                duration={getDurationTime(contract_info)}
-                duration_unit={getDurationUnitText(getDurationPeriod(contract_info))}
-                exit_spot={exit_spot}
-                has_result={!!is_sold || is_multiplier}
-                toggleHistoryTab={toggleHistoryTab}
-            />
-        );
-
-        const body_content = (
-            <React.Fragment>
-                {this.props.contract_info.status || fallback_result ? (
-                    this.getBodyContent()
-                ) : (
-                    <div className='contract-card'>
-                        <PositionsCardLoader is_dark_theme={this.props.is_dark_theme} speed={2} />
-                    </div>
-                )}
-            </React.Fragment>
-        );
-
-        const contract_drawer = (
-            <CSSTransition
-                in={this.state.should_show_contract_audit}
-                timeout={250}
-                classNames='contract-drawer__transition'
+    if (!contract_info) return null;
+
+    // For non-binary contract, the status is always null, so we check for is_expired in contract_info
+    const fallback_result = contract_info.is_expired;
+
+    const exit_spot = isUserSold(contract_info) && !is_multiplier ? '-' : exit_tick_display_value;
+
+    const contract_audit = (
+        <ContractAudit
+            contract_info={contract_info}
+            contract_update_history={contract_update_history}
+            contract_end_time={getEndTime(contract_info)}
+            is_dark_theme={is_dark_theme}
+            is_multiplier={is_multiplier}
+            is_open
+            duration={getDurationTime(contract_info)}
+            duration_unit={getDurationUnitText(getDurationPeriod(contract_info))}
+            exit_spot={exit_spot}
+            has_result={!!is_sold || is_multiplier}
+            toggleHistoryTab={toggleHistoryTab}
+        />
+    );
+
+    const body_content = (
+        <React.Fragment>
```

Comment:
@amirghadimi-fs I think we don't need ` <React.Fragment>` here

Reply:
We need it Shayan, because we have `{` and a condition not a simple tag.

Reply:
well it's an if a condition so could we have thing like `(contract_info.status || fallback_result) ? getBodyContent() : (<div className='contract-card'><PositionsCardLoader is_dark_theme={is_dark_theme} speed={2} /></div>)`